orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -442,27 +491,30 @@ def _get_forward_graph(self, input_names, dynamic_axes, *inputs, **kwargs):
         TODO: How to support dynamic axes? Dimensions are determined by samples
         TODO: How to ingest **kwargs in proper order during export?
         '''
-        # Export the model to memory
-        f = io.BytesIO()
 
-        # Deepcopy inputs, since input values may change after model run.
-        # NOTE: Inputs may contain tensors that have attributes preventing their deepcopy (example grad_fn).
-        # Therefore, deepcopy only the data component of the input tensors for export.
-        sample_inputs_copy = []
-        for model_input in inputs:
-            sample_inputs_copy.append(model_input.data if isinstance(model_input, torch.Tensor) else model_input)
-        sample_inputs_copy = copy.deepcopy(tuple(sample_inputs_copy))
+        # Setup dynamic axes for onnx model
+        input_names, dynamic_axes, _ = _parse_inputs_for_onnx_export(self._original_module, *inputs, **kwargs)
```

Comment:
The arguments to this function have already been through ```_parse_inputs_for_onnx_export``` function before the ```_get_forward_graph``` function is invoked.
We could remove the call to ```_parse_inputs_for_onnx_export``` before this function is called (also assigning a value to ```self._input_names_requires_grad``` inside this function.
In that way, we don't need to pass in all these arguments: ```input_names``` and ```dynamic_axes```

Reply:
My understanding from the requires_grad PR is that `_input_names_requires_grad` needed to be evaluated for every iteration to make sure the user didnt change requires_grad of any input.